### PR TITLE
Adding Physical Switches support

### PR DESCRIPTION
**Pull request by saulotoledo**
Add Physical Switches to the API.

Depends on:
- ~https://github.com/ManageIQ/manageiq/pull/17409~

### CODE REVIEW

**Comment by saulotoledo** [comment]
@abellotti @imtayadeway Could you review this PR?

**Comment by saulotoledo** [comment]
@miq-bot add_reviewer @abellotti @imtayadeway

**Comment by saulotoledo** [comment]
@miq-bot review @imtayadeway

**Comment by saulotoledo** [comment]
@miq-bot add_reviewer @imtayadeway

File: spec/requests/physical_switches_spec.rb [hunk 1]
```diff
@@ -0,0 +1,100 @@
+describe "Physical Switches API" do
+  context "GET /api/physical_switch" do
```
Comment (abellotti) [inline spec/requests/physical_switches_spec.rb]: /api/physical_switches

File: spec/requests/physical_switches_spec.rb [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    it "returns all physical_switch" do
```
Comment (abellotti) [inline spec/requests/physical_switches_spec.rb]: physical_switches

File: spec/requests/physical_switches_spec.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        "name"      => "physical_switches",
+        "resources" => [{"href" => api_physical_switch_url(nil, physical_switch)}]
+      }
+      expect(response).to have_http_status(:ok)
+      expect(response.parsed_body).to include(expected)
+    end
+  end
+
+  context "GET /api/physical_switch/:id" do
+    it "returns one physical_switch" do
```
Comment (abellotti) [inline spec/requests/physical_switches_spec.rb]: same changes to reflect physical_switches

File: spec/requests/physical_switches_spec.rb [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+        expect(response).to have_http_status(:forbidden)
+      end
+    end
+
+    context "with an appropriate role" do
+      it "rejects refresh for unspecified physical switch" do
+        api_basic_authorize(action_identifier(:physical_switches, :refresh, :resource_actions, :post))
+
+        post(api_physical_switches_url, :params => gen_request(:refresh, [{"href" => "/api/physical_switches/"}, {"href" => "/api/physical_switches/"}]))
```
Comment (abellotti) [inline spec/requests/physical_switches_spec.rb]: maybe use api_physical_switches_url  helper method for the above 2 href.

File: spec/requests/physical_switches_spec.rb (hunk 1 above)
Comment (saulotoledo) [reply spec/requests/physical_switches_spec.rb]: ops... done.

File: spec/requests/physical_switches_spec.rb (hunk 2 above)
Comment (saulotoledo) [reply spec/requests/physical_switches_spec.rb]: done

File: spec/requests/physical_switches_spec.rb (hunk 3 above)
Comment (saulotoledo) [reply spec/requests/physical_switches_spec.rb]: done

File: spec/requests/physical_switches_spec.rb (hunk 4 above)
Comment (saulotoledo) [reply spec/requests/physical_switches_spec.rb]: done

**Comment by abellotti** [comment]
Thanks @saulotoledo for the updates and the API Enhancement. LGTM !! 👍

**Pull request merged**